### ISSUE TITLE
Using allOf in array leads to NullPointerException: property must not be null

### ISSUE THREAD

**Issue by meyeraxel** [Swagger2Markup/swagger2markup#155]
Swagger2Markup version: 1.0.0

It seems like using the allOf keyword in an array causes a NullPointerException. Here is an isolated swagger file which causes the issue:
```
---
swagger: '2.0'
info:
  version: 0.0.0
  title: Simple API
paths:
  /:
    get:
      responses:
        200:
          description: OK
          schema:
            $ref: "#/definitions/MyObject"
definitions:
  HATEOAS_links:
    description: The link object provieds links to ressources which are related to the response
    required:
      - "@type"
      - rel
      - uri
    properties:
      "@type":
        type: string
        default: Link
        description: This is the @type definition for this reference object.
      rel:
        type: string
        default: self
        description: Defines the relation from the object to the link. The minimum response is the self link.
      uri:
        type: string
        format: uri
        description: The URL to the ressource which is related
  MyObject:
    properties:
      isActive:
        type: boolean
      links:
        type: array
        items:
          allOf:
            - $ref: "#/definitions/HATEOAS_links"
            - properties:
                rel:
                  type: string
                  default: owner
                  description: Reference to the owner of the object
            - properties:
                rel:
                  type: string
                  default: customer
                  description: Reference to the customer of the object
```

Problem description:
```
java -jar swagger2markup-cli-1.0.0.jar convert -i swagger.yaml -f test
09:50:16.116 [main] INFO  io.swagger.parser.Swagger20Parser - reading from C:/temp/swagger2markup/swagger.yaml
09:50:16.319 [main] DEBUG i.s.m.properties.PropertyBuilder - no property for null, null
09:50:16.319 [main] WARN  io.swagger.util.PropertyDeserializer - no property from null, null, {ENUM=null, TITLE=null, DESCRIPTION=null, DEFAULT=null, PATTERN=null, DESCRIMINATOR=null, MIN_ITEMS=null,
MAX_ITEMS=null, MIN_PROPERTIES=null, MAX_PROPERTIES=null, MIN_LENGTH=null, MAX_LENGTH=null, MINIMUM=null, MAXIMUM=null, EXCLUSIVE_MINIMUM=null, EXCLUSIVE_MAXIMUM=null, UNIQUE_ITEMS=null, EXAMPLE=null
 TYPE=null, FORMAT=null, READ_ONLY=null, VENDOR_EXTENSIONS={}}
09:50:16.413 [main] INFO  i.g.s.m.b.i.asciidoc.AsciiDocBuilder - Markup document written to: C:\temp\swagger2markup\test.adoc
09:50:16.413 [main] DEBUG i.g.s.i.d.b.PathsDocumentBuilder - Include examples is disabled.
09:50:16.413 [main] DEBUG i.g.s.i.d.b.PathsDocumentBuilder - Create separated operation files is disabled.
09:50:16.584 [main] INFO  i.g.s.i.d.b.PathsDocumentBuilder - Operation processed : '/ get' (normalized id = 'get')
09:50:16.600 [main] INFO  i.g.s.m.b.i.asciidoc.AsciiDocBuilder - Markup document written to: C:\temp\swagger2markup\test.adoc
09:50:16.600 [main] DEBUG i.g.s.i.d.b.DefinitionsDocumentBuilder - Create separated definition files is disabled.
09:50:16.615 [main] INFO  i.g.s.i.d.b.DefinitionsDocumentBuilder - Definition processed : 'HATEOAS_links'
Exception in thread "main" java.lang.NullPointerException: property must not be null
        at org.apache.commons.lang3.Validate.notNull(Validate.java:222)
        at io.github.swagger2markup.internal.utils.PropertyUtils.getType(PropertyUtils.java:43)
        at io.github.swagger2markup.internal.utils.PropertyUtils.getType(PropertyUtils.java:54)
        at io.github.swagger2markup.internal.document.builder.MarkupDocumentBuilder.buildPropertiesTable(MarkupDocumentBuilder.java:182)
        at io.github.swagger2markup.internal.document.builder.DefinitionsDocumentBuilder.typeSection(DefinitionsDocumentBuilder.java:264)
        at io.github.swagger2markup.internal.document.builder.DefinitionsDocumentBuilder.buildDefinition(DefinitionsDocumentBuilder.java:197)
        at io.github.swagger2markup.internal.document.builder.DefinitionsDocumentBuilder.buildDefinition(DefinitionsDocumentBuilder.java:171)
        at io.github.swagger2markup.internal.document.builder.DefinitionsDocumentBuilder.buildDefinitionsSection(DefinitionsDocumentBuilder.java:110)
        at io.github.swagger2markup.internal.document.builder.DefinitionsDocumentBuilder.build(DefinitionsDocumentBuilder.java:97)
        at io.github.swagger2markup.Swagger2MarkupConverter.toFile(Swagger2MarkupConverter.java:207)
        at io.github.swagger2markup.cli.Application.run(Application.java:84)
        at io.github.swagger2markup.cli.Application.main(Application.java:61)
```

**Comment by Kabhal** [comment]
I think this is a duplicate of https://github.com/Swagger2Markup/swagger2markup/issues/150 
Problem should be solved on master branch (1.0.1-SNAPSHOT)
Please let me know.

**Comment by meyeraxel** [comment]
Yes you are right, it is a duplicate and it is running with version 1.0.1-20160601.130400-11 without a problem. I think this issue can be closed, sorry for adding this twice.
